code/game/spacepods/construction.dm

```diff
@@ -0,0 +1,229 @@
+/obj/structure/spacepod_frame
+	density = 1
+	opacity = 0
+	bound_width = 64
+	bound_height = 64
+	anchored = 1
+	layer = 3.9
+	name = "space pod frame"
+	icon = 'icons/48x48/pod_construction.dmi'
+	icon_state = "pod_1"
+
+	var/datum/construction/construct
+
+/obj/structure/spacepod_frame/New()
+	..()
+
+	construct = new /datum/construction/reversible/pod(src)
+
+	dir = EAST
+
+/obj/structure/spacepod_frame/attackby(obj/item/W as obj, mob/user as mob)
+	if(!construct || !construct.action(W, user))
+		return ..()
+	return 1
+
+/obj/structure/spacepod_frame/attack_hand()
+	return
+
+/datum/construction/reversible/pod/spawn_result(mob/user as mob)
+	..()
+	feedback_inc("spacepod_created",1)
+	return
+
+/datum/construction/reversible/pod/action(atom/used_atom,mob/user)
+	return check_step(used_atom,user)
+
+/datum/construction/reversible/pod/custom_action(index as num, diff as num, atom/used_atom, mob/user)
+	if(istype(used_atom, /obj/item/weapon/weldingtool))
```

Comment:
you can't ```switch(type)``` anyway, because that means it ignores all subtypes of the provided path:
```
switch(type)
        if(/obj/item/weapon/weldingtool) //Will not find subtypes.
        if(/obj/item/weapon/weldingtool, /obj/item/weapon/weldingtool/largetank, /obj/item/weapon/weldingtool/experimental, ...) //Not maintainable, Not acceptable
```